pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -176,6 +204,11 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 		return r.reconcileAndError(instance, err)
 	}
 
+	err = r.obtainLease(node)
+	if err != nil {
+		return r.reconcileAndError(instance, err)
```

Comment:
I don't think this is always the correct response to an error.
For example, if we still have the lease but cannot extend it, we would want to call `stopNodeMaintenance()`

Reply:
and what should happen if we fail to update the  lease is still valid, but about to expire within grace period?

Reply:
ok, but then you also need to ad a NodePhase ; currently you have "Running" and "Succeeded" so you need a "Failed" state as well. then you will have to make sure that this is also displayed in the GUI and whoever is using this controller besides the gui.

Reply:
done

Reply:
fixed it again, according to clarifications received.